src/java.desktop/share/native/libharfbuzz/hb.hh

```diff
@@ -129,6 +131,7 @@
 #pragma GCC diagnostic ignored "-Wclass-memaccess"
 #pragma GCC diagnostic ignored "-Wcast-function-type-strict" // https://github.com/harfbuzz/harfbuzz/pull/3859#issuecomment-1295409126
 #pragma GCC diagnostic ignored "-Wdangling-reference" // https://github.com/harfbuzz/harfbuzz/issues/4043
+#pragma GCC diagnostic ignored "-Wdangling-pointer" // Trigerred by hb_decycler_node_t().
```

Comment:
@prrace With harfbuzz v10.4.0, dangling-pointer warning is ignored. With this addition, I assumed that we no longer need to add it to harfbuzz warning section explicitly. 

But without adding dangling-pointer to `HARFBUZZ_DISABLED_WARNINGS_gcc` build fails on linux so probably JDK build requires warnings to be explicitly added to ClientLibraries.gmk ?

```
decycler.hh:110:25: error: storing the address of local variable ‘decycler_node'
 in '((hb_decycler_t*)((char*)scratch + 8))[1].hb_decycler_t::tortoise' [-Werror=dangling-pointer=]
  110 |       decycler.tortoise = decycler.hare = this;
      |       ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Interesting. I also would have expected this to over-ride the JDK default.
But if that's what you see ... 
A bit  of googling suggests that sometimes these pragmas are read too late.